examples/retry-backoff.yaml

```diff
@@ -0,0 +1,20 @@
+# This example demonstrates the use of retry back offs
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: retry-backoff-
+spec:
+  entrypoint: retry-backoff
+  templates:
+  - name: retry-backoff
+    retryStrategy:
+      limit: 10
+      backoff:
+        duration: 1       # Default unit is seconds. Could also be a Duration, e.g.: "2m", "6h", "1d"
+        factor: 2
```

Comment:
Sorry for the bikeshedding, but should this not be called "base" instead, given that it is, in fact, the base of the exponential function? 
Also, good work :-)

Reply:
Duration is actually [what K8s uses](https://github.com/kubernetes/kubernetes/blob/c97baa339b5043424522dc7932fc3891f15543cb/staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go#L206-L230) :) I think because technically `factor` is optional; if left at `0` then the backoff would simply apply the same `duration` wait